src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -72,6 +72,12 @@
     @Query("select distinct course from Course course where (course.startDate is null or course.startDate <= :#{#now}) and (course.endDate is null or course.endDate >= :#{#now}) and course.onlineCourse = false and course.registrationEnabled = true")
     List<Course> findAllCurrentlyActiveNotOnlineAndRegistrationEnabled(@Param("now") ZonedDateTime now);
 
+    @Query("select distinct course from Course course left join fetch course.organizations co where (course.startDate is null or course.startDate <= :#{#now}) and (course.endDate is null or course.endDate >= :#{#now}) and course.onlineCourse = false and course.registrationEnabled = true")
+    List<Course> findAllCurrentlyActiveNotOnlineAndRegistrationEnabledWithOrganizations(@Param("now") ZonedDateTime now);
+
+    @Query("select course from Course course left join fetch course.organizations co where course.id = :#{#courseId}")
+    Course findWithEagerOrganizations(@Param("courseId") long courseId);
```

Comment:
More modern here would be to return Optional<Course> but just a suggestion.

Reply:
You are right, i missed a 'findWithEagerOrganizations**ElseThrow**' method which returns the Course entity directly.

Added Optional and the additional method in 7507bfa